### ISSUE TITLE
Misleading documentation for `Model.predict()` compared to `model()`

### ISSUE THREAD

**Issue by lebigot** [keras-team/keras#15186]
**URL(s) with the issue:**.

https://www.tensorflow.org/api_docs/python/tf/keras/Model#predict

**Description of issue (what needs to be changed):**.

The documentation for `predict()` implies that it gives the same output as `__call__()`:

> For small amount of inputs that fit in one batch, directly using `__call__` is recommended for faster execution, e.g., model(x)…

A warning is in order here: `__call__()` returns TensorFlow tensors (whereas `predict()` returns NumPy arrays). We would for instance write instead (added text in bold):

> For small amount of inputs that fit in one batch, directly using `__call__` is recommended for faster execution, e.g., `model(x)`, or `model(x, training=False)` if you have layers such as `tf.keras.layers.BatchNormalization` that behaves differently during inference **(note that `__call__` returns tensors instead of arrays)**.

**Comment by jvishnuvardhan** [comment]
@lebigot Agree with you. Output value from `model.predict(x)` and model(x) is identical but in different format (`array` versus `tf.Tensor`). 

You can get same format by 

```
# predictions = model(x_train[:1]) # returns tf.Tensor
predictions = model(x_train[:1]).numpy() # returns array
```

Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/d30d7504c57a0c80228d9a979205ba58/untitled.ipynb)

Please let me know if you want to open a PR to update the docs.? If not, I can update it through PR. Thanks!

**Comment by lebigot** [comment]
If you can indeed open a PR, that'd be great. Thanks!
